src/segger/data/io.py

```diff
@@ -272,7 +309,7 @@ def set_metadata(self) -> None:
         instead of one-hot encodings and store the lookup table for later mapping.
         """
         # Load the Parquet file metadata
-        parquet_file = pq.ParquetFile(self.transcripts_path)
+        parquet_file = pq.read_table(self.transcripts_path)
```

Comment:
Using `read_table` allows to read multi-file parquet files, like the one that are used in `spatialdata` and that arise from a Dask `DataFrame` that has many partitions.

I didn't do a proper benchmark, and just tried the code below on the Xenium pancreas dataset. It takes 10 seconds instead of 8 with the new implementation. I'm sure it can be made more efficient.